pkg/labels/selector.go

```diff
@@ -72,10 +72,13 @@ const (
 	ExistsOperator       Operator = "exists"
 )
 
-//LabelSelector is a list of Requirements.
-type LabelSelector []Requirement
+func NewSelector() Selector {
```

Comment:
Ya, I think that Add should at least be refactored to take a single Requirement instead of the three args it takes now. I'm so so on whether it should be variadic.